### PR TITLE
feat: run post_hooks even if parent nodes fail

### PR DESCRIPTION
**Pull request by mederka**

### CODE REVIEW

File: src/fal/cli/flow_runner.py [hunk 1]
```diff
@@ -102,31 +102,47 @@ def _run_sub_graph(
...
-            raise_for_run_results_failures(python_node_scripts, results)
 
         if post_hooks:
             results = run_scripts(post_hooks, fal_dbt)
             raise_for_run_results_failures(post_hooks, results)
 
-    if len(after_scripts) != 0:
+        raise_for_dbt_run_errors(output)
+
+        if fal_nodes:
```
Comment (chamini2) [inline src/fal/cli/flow_runner.py:112]: This cannot be moved after post_hooks. This is where we run Python models.

Probably need to do the `_mark_dbt_nodes_status` again.

**Review by chamini2 — CHANGES_REQUESTED** [review]

File: src/fal/cli/flow_runner.py [hunk 2]
```diff
@@ -102,31 +102,65 @@ def _run_sub_graph(
...
+                        fal_dbt,
+                        NodeStatus.Success,
+                        python_node_scripts[i].model.unique_id,
+                    )
+                else:
+                    _mark_dbt_nodes_status(
+                        fal_dbt,
+                        NodeStatus.Error,
+                        python_node_scripts[i].model.unique_id,
+                    )
```
Comment (chamini2) [inline src/fal/cli/flow_runner.py:124]:
```suggestion
            for script, success in zip(python_node_scripts, results):
                if success:
                    _mark_dbt_nodes_status(
                        fal_dbt,
                        NodeStatus.Success,
                        script.model.unique_id,
                    )
                else:
                    _mark_dbt_nodes_status(
                        fal_dbt,
                        NodeStatus.Error,
                        script.model.unique_id,
                    )
```

**Review by chamini2 — APPROVED** [review]
Cosmetic details. Good one!

File: src/fal/cli/flow_runner.py (hunk 2 above)
Comment (chamini2) [reply src/fal/cli/flow_runner.py:124]: Even
```suggestion
            for script, success in zip(python_node_scripts, results):
                  _mark_dbt_nodes_status(
                      fal_dbt,
                      NodeStatus.Success if success else NodeStatus.Error,
                      script.model.unique_id,
                  )
```

Comment (mederka) [reply src/fal/cli/flow_runner.py:124]: ❤️

**Pull request merged**